### ISSUE TITLE
twitch_vod_fetch does not finish appending batched chunks

### ISSUE THREAD

**Issue by X1011** [mk-fg/fgtk#17]
The script often does not append the last batch of chunks to the destination file. This causes the VoD to be truncated up to 100 seconds early with the default config. Consider the following example. 
For ease of demonstration, I changed line 27 in my local copy to `file_append_batch = 2`. twitch_vod_fetch output truncated for legibility
```
* $ twitch_vod_fetch https://www.twitch.tv/videos/347546662 10 -l 10 -k
  ------ Started 1 downloads, last gid: 000001 ------  
* $ twitch_vod_fetch https://www.twitch.tv/videos/347546662 40 -l 40 -k
  ------ Started 4 downloads, last gid: 000004 ------  
* $ twitch_vod_fetch https://www.twitch.tv/videos/347546662 5chunks-by2 -l 50 -k
  ------ Started 5 downloads, last gid: 000005 ------  
```
40.mp4 comes out fine (and for some reason, when I tried 20- and 30-second videos, they worked, too), but 5chunks-by2.mp4's last chunk is left in the tmp folder, and poor 10.mp4 never receives its single chunk!
```
* $ ls -l 5chunks-by2.mp4 40.mp4 10.mp4 
-rw-r--r-- 1 dan dan        0 Jan 14 07:01 10.mp4
-rw-r--r-- 1 dan dan 15039812 Jan 14 07:04 40.mp4
-rw-r--r-- 1 dan dan 15039812 Jan 14 07:10 5chunks-by2.mp4
* $ ls 5chunks-by2.tmp/
5chunks-by2.0000050000000000.mp4.chunk	5chunks-by2.aria2c_conf  5chunks-by2.m3u8  5chunks-by2.m3u8.ua	5chunks-by2.m3u8.url  5chunks-by2.pos
* $ ls 40.tmp/
40.aria2c_conf	40.m3u8  40.m3u8.ua  40.m3u8.url  40.pos
* $ ls 10.tmp/
10.0000010000000000.mp4.chunk  10.aria2c_conf  10.m3u8	10.m3u8.ua  10.m3u8.url  10.pos
```
I've seen this issue on many (longer) videos since the batched chunk append was implemented; I just chose this short highlight as an example.
Thanks again for your work :)

**Comment by mk-fg** [comment]
Ah yes, I was wondering why I was seeing videos seemingly cut a little short for a while, but never thought to check, always thinking "eh, must be twitch anyway". Will look into it in a moment.

**Comment by mk-fg** [comment]
Pretty sure it was missing chunks.finished check, now fixed in 7214985, but I can't seem to reproduce the issue so far, which is weird, as it definitely should happen with that check missing, will try a bit more.

**Comment by mk-fg** [comment]
No, can totally reproduce it even more obviously with "file_append_batch = 10" left as it is and 5chunks-by2 command, where resulting file is left completely empty and all chunks in the tmp dir.
It's likely due to arbitrary order in which aria2 finished downloading chunks in parallel, while script has to append them in strict order.

Anyway, that missing check was definitely an issue and skipped appending last 0-file_append_batch chunks.
If you got lucky and download ended with "n >= file_append_batch" chunks still in tmp, it'd append them, which is why cutting short only happened in some cases, but not all of them.
Can happen due to that random downloading order ([re-]started sequentially, finish whenever), e.g. if chunk-1 out of 20 finishes last, all 20 would've been appended after that just fine.

Should be fixed now, should've looked into it earlier myself.
Thanks a lot for reporting!
